### ISSUE TITLE
Slow connections every startup... Would hibernate help?

### ISSUE THREAD

**Issue by Xuchilbara** [BiglySoftware/BiglyBT#3236]
Hi,

BiglyBT tells me each and every time I start it that trackers are lagging and it certainly takes a good 10, 20, 30... minutes to get the connections and all to start working.

I tried everything, but it just keeps the same, I don't know if it could be due to many torrents.

But what I was wondering is, would hibernate the computer help? I mean, if the state of BiglyBT is saved to the HDD, then the next time I turn on the PC it should take less time to start working properly, isn't it? Or maybe not?

So, can hibernating help speed things up at start with P2P/torrents or the connections are lost anyway or something? I couldn't find an answer to this anywhere.

* OS and version: Windows 10 Pro x64 22H2 with all up-to-date.
* BiglyBT Version Number: 3.5.0.0.

Thank you!

**Comment by parg** [comment]
If trackers are lagging you are running too many active torrents given your tracker concurrency settings - as it says in the message I believe.,..

Check Stats->Tracker and "Options->Tracker->Client: Maximum concurrent announce tasks"

You'd have to experiment with hibernate, not sure it will help.

**Comment by Xuchilbara** [comment]
What is a good number for Maximum concurrent announce tasks? I have 25 and I don't remember ever touching it.

Thank you!

**Comment by Xuchilbara** [comment]
I tried hibernation and it seems to be working well, it starts way faster and as soon as the OS is working Bigly starts downloading and seeding right away almost as it was left hours ago.

The downside is my hibernation file, hiberfil.sys, takes like 12 GB of disk space considering I have 16 GB of RAM in this computer and a pagefile of 4 GB. Imagine if one has 32 or 64 GB of RAM!

As for the stats... I don't know what I'm looking for. It says Update Rates 1.95/s or 06/s... Update Lag 742 s, Scheduled 2224/1390...

Thanks.

**Comment by parg** [comment]
Update lag of 742s - means you are running too many active torrents with too low "Options->Tracker->Client: Maximum concurrent announce tasks". You may also have misbehaving trackers in your torrents that are causing these kind of issues. 

Go to View->All Trackers, enable the "latency" column and see if there are trackers stalling things

**Comment by Xuchilbara** [comment]
According to Latency the biggest number is 68773 ms and the second one 28552 ms, then four more around the 20k, then couple around 10k and the rest are below 3k.

**Comment by parg** [comment]
if the high latency ones aren't producing any useful results (e.g. timing out with errors) then remove them

**Comment by Xuchilbara** [comment]
Some of them say "Offline (No data received from tracker)", sometimes some "OK".

I'll try to remove the offline trackers from the torrents and see how it goes.

Should I increase the concurrency number from the 25 I have then or is it better to leave it?

As for hibernation, it does seem to retain at least most connections (after around 12 h), of course not all of them because peers also go offline. The only problem I had is something from the system waking the computer, but I hope I solved that. Other than the extra disk space is worth it for the time saved, if it doesn't give other problems.

Thank you!

**Comment by parg** [comment]
Sure, increase the concurrency, just don't go too crazy!

**Comment by Xuchilbara** [comment]
So I tried to remove the bad trackers... but I kinda messed up editing them together since there were more than 400 torrents and so many trackers, so I only made the problem worse and the update lag was around 6000 s and going up and the scheduled around 16 000... Then I just removed them all and added the five best ones I had listed in All Trackers.

Then restarted. So now the update rates graph is around 1.25/s (oscillating between 0.5 and 2.25 or so), the update lag is 0 s and, the scheduled around 700. And no more lagging message at last!

I also increased the concurrency to 50.

So finally I think the "problem" is solved and I've learned more about Bigly and just P2P or torrent in general. So thanks a lot!
